### PR TITLE
feat(meta): Add V2 client for querying device profile

### LINKED ISSUE #392: [V2 meta] Add V2 client for querying deviceprofile
Add V2 client for querying the deviceprofile 

- [query by name](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/default/get_deviceprofile_name__name_)
- [query all](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/default/get_deviceprofile_all)
- [query all by model](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/default/get_deviceprofile_model__model_)
- [query all by manufacturer](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/default/get_deviceprofile_manufacturer__manufacturer_)
- [query all by manufacturer and model](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/default/get_deviceprofile_manufacturer__manufacturer__model__model_)

### PR DESCRIPTION
**Pull request by weichou1229**
Fix #392

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
Add V2 client for querying the device profile
- query by name
- query all
- query all by model
- query all by manufacturer
- query all by manufacturer and model

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: v2/clients/http/deviceprofile.go [hunk 1]
```diff
@@ -44,3 +49,87 @@ func (client *DeviceProfileClient) Update(ctx context.Context, reqs []requests.D
...
+	u.Path = path.Join(u.Path, v2.ApiDeviceProfileRoute, v2.Manufacturer, url.QueryEscape(manufacturer), v2.Model, url.QueryEscape(model))
+	q := u.Query()
+	q.Set(v2.Offset, strconv.Itoa(offset))
+	q.Set(v2.Limit, strconv.Itoa(limit))
+	u.RawQuery = q.Encode()
+	err = utils.GetRequest(ctx, &res, u.String())
+	if err != nil {
+		return res, errors.NewCommonEdgeXWrapper(err)
+	}
+	return res, nil
```
Comment (lenny-intel) [inline v2/clients/http/deviceprofile.go:134]: This code is repeated (except for the parameters to Join) in most of these functions, . Please refactor into helper function that takes var args for the Join parameters.

File: v2/clients/http/deviceprofile.go [hunk 2]
```diff
@@ -44,3 +49,87 @@ func (client *DeviceProfileClient) Update(ctx context.Context, reqs []requests.D
...
+	u, err := url.Parse(client.baseUrl)
+	if err != nil {
+		return res, errors.NewCommonEdgeX(errors.KindClientError, "fail to parse baseUrl", err)
+	}
+	u.Path = path.Join(u.Path, v2.ApiDeviceProfileRoute, v2.Name, url.QueryEscape(name))
+	err = utils.GetRequest(ctx, &res, u.String())
+	if err != nil {
+		return res, errors.NewCommonEdgeXWrapper(err)
+	}
+	return res, nil
```
Comment (lenny-intel) [inline v2/clients/http/deviceprofile.go:63]: Refactor into helper function like in other comment but without query params. Will be useful as out GETs are added.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: v2/clients/http/deviceprofile.go (hunk 1 above)
Comment (weichou1229) [reply v2/clients/http/deviceprofile.go]: Refactored the GetRequest function to accept the request path and params.

File: v2/clients/http/deviceprofile.go (hunk 2 above)
Comment (weichou1229) [reply v2/clients/http/deviceprofile.go]: Refactored.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**